test/core.jl

```diff
@@ -5649,3 +5649,9 @@ function hh6614()
     x, y
 end
 @test hh6614() == (1, 2)
+
+# issue 22098
+macro m22098 end
+handle_on_m22098 = getfield(@__MODULE__, Symbol("@m22098"))
+methods_m22098 = methods(handle_on_m22098)
+@test isempty(methods_m22098)
```

Comment:
No need for the intermediate variables here.

Reply:
I find it more readable like this. What is the benefit of spelling it 
```julia
@test isempty(methods(getfield(@__MODULE__, Symbol("@m22098"))))
```
?

Reply:
I agree on removing the `methods_m22098` variable though.